src/platform/src/main/java/org/geoserver/platform/GeoServerEnvironment.java

```diff
@@ -97,29 +103,57 @@ public Properties getProps() {
 
     public GeoServerEnvironment() {
         try {
-            GeoServerResourceLoader loader =
-                    GeoServerExtensions.bean(GeoServerResourceLoader.class);
-            configFile =
-                    new FileWatcher<Properties>(loader.get(PROPERTYFILENAME)) {
-
-                        @Override
-                        protected Properties parseFileContents(InputStream in) throws IOException {
-                            Properties p = new Properties();
-                            p.load(in);
-                            return p;
-                        }
-                    };
-
+            this.configFile = getFromPathOrDataDir(propertiesPath);
             props = configFile.read();
         } catch (Exception e) {
-            LOGGER.log(
-                    Level.WARNING,
-                    "Could not find any '" + PROPERTYFILENAME + "' property file.",
-                    e);
+            String file = propertiesPath != null ? propertiesPath : PROPERTYFILENAME;
+            LOGGER.log(Level.WARNING, "Could not find any '" + file + "' property file.", e);
             props = new Properties();
         }
     }
 
+    private FileWatcher<Properties> getFromPathOrDataDir(String propertiesFilePath) {
```

Comment:
Name after what a method does, not based on how it does it (we might extend it later to load from a S3 bucket, for what I know).

```suggestion
    private FileWatcher<Properties> getEnvironmentProperties(String propertiesFilePath) {
```

Reply:
fixed